test/livebook_web/live/session_live_test.exs

```diff
@@ -913,16 +913,30 @@ defmodule LivebookWeb.SessionLiveTest do
     end
   end
 
-  describe "add secret" do
+  describe "secrets" do
     test "adds a secret from form", %{conn: conn, session: session} do
       {:ok, view, _} = live(conn, "/sessions/#{session.id}/secrets")
 
       view
       |> element(~s{form[phx-submit="save"]})
-      |> render_submit(%{data: %{label: "foo", value: "123"}})
+      |> render_submit(%{secret: %{label: "foo", value: "123"}})
 
       assert %{secrets: [%{label: "FOO", value: "123"}]} = Session.get_data(session.pid)
     end
+
+    test "shows the 'Add secret' button for unavailable secrets", %{conn: conn, session: session} do
+      section_id = insert_section(session.pid)
+      cell_id = insert_text_cell(session.pid, section_id, :code, ~s{System.fetch_env!("LB_FOO")})
+
+      Session.queue_cell_evaluation(session.pid, cell_id)
+      wait_for_session_update(session.pid)
```

Comment:
@cristineguadelupe it's a race condition, we may render the page before the session gets the evaluation result. The `wait_for_session_update` function only makes sure the session processes all it's current messages. We can explicitly wait for the evaluation to finish:

```suggestion
      Session.subscribe(session.id)
      section_id = insert_section(session.pid)
      cell_id = insert_text_cell(session.pid, section_id, :code, ~s{System.fetch_env!("LB_FOO")})

      Session.queue_cell_evaluation(session.pid, cell_id)
      assert_receive {:operation, {:add_cell_evaluation_response, _, ^cell_id, _, _}}
```